### PR TITLE
DBAAS1-80: Database Detail Data Scaffolding

### PR DESCRIPTION
**Pull request by bnussman**
## Description

- Sources data for Database Details and Summary with React Query
- Following PRs for DBaaS are bound to make changes to this initial scaffolding
- Fixed an incorrect type in `@linode/api-v4`
- Added **MSW** mock for `/databases/:engine/instances/:id`

## How to test

- Go to a Database details page (`/databases/3` for example) and make sure you see JSON data.

Screen Shot 2021-12-22 at 12 05 50 PM

### CODE REVIEW

**Comment by bnussman** [comment]
Requesting review from @tiffwong @bcmertz @HanaXu because you guys will be working on the Database Details pages

**Review by DevDW — APPROVED** [review]
LGTM, left a couple of comments and a question

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseEntityDetail.tsx [hunk 1]
```diff
@@ -26,32 +26,40 @@ import InlineTextLoader from 'src/components/InlineTextLoader';
...
+  const id = Number(databaseId);
+  const engine = getDatabaseEngine(id);
 
-const DatabaseEntityDetail: React.FC<DatabaseEntityDetailProps> = (props) => {
-  const { database } = props;
+  const { mutateAsync: deleteDatabase } = useDeleteDatabaseMutation(engine, id);
+  const { data: database, isLoading, error } = useDatabaseQuery(engine, id);
+  const { data: types } = useDatabaseTypesQuery();
+
+  const type = types?.find((type) => type.id === database?.type);
```
[the remark this answers is not in the document]
Comment (bnussman) [reply packages/manager/src/features/Databases/DatabaseDetail/DatabaseEntityDetail.tsx:52]: From looking at the API spec, it looks like the type will be a string, which is why I did the `find` for the type data. Is that what you see in the spec? @DevDW

**Review by tiffwong — APPROVED** [review]
🚀

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseEntityDetail.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (bcmertz) [reply packages/manager/src/features/Databases/DatabaseDetail/DatabaseEntityDetail.tsx:47]: it looks like as @DevDW mentioned the spec returns a Database object which has `type` of `linode type slug` which corresponds to the frontend type of `DatabaseType`. From which we could get the `id` field

This is from the spec for POST /databases/{engine}/instances

**Review by bcmertz — APPROVED** [review]
LGTM pending resolution of @DevDW 's feedback! 🎆

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseEntityDetail.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (bnussman) [reply packages/manager/src/features/Databases/DatabaseDetail/DatabaseEntityDetail.tsx:47]: I guess my question now is, is `type` (`linode type slug`) from `/databases/:engine/instances/:id` a string or an object?

[the remark this answers is not in the document]
Comment (bcmertz) [reply packages/manager/src/features/Databases/DatabaseDetail/DatabaseEntityDetail.tsx:47]: yep it's a string that's used for lookups according to API team

[the remark this answers is not in the document]
Comment (bcmertz) [reply packages/manager/src/features/Databases/DatabaseDetail/DatabaseEntityDetail.tsx:47]: in this case I believe it can only be "database" I believe

**Pull request merged**